### PR TITLE
Update buf CLI to 1.32 + migrate

### PR DESCRIPTION
**Pull request by rodaine**

### CODE REVIEW

File: buf.gen.yaml [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+    - file_option: go_package
+      module: buf.build/envoyproxy/protoc-gen-validate
+    - file_option: go_package
+      module: buf.build/bufbuild/protovalidate
+  override:
+    - file_option: go_package_prefix
+      value: github.com/bufbuild/protovalidate-go/internal/gen
 plugins:
-  - plugin: buf.build/protocolbuffers/go:v1.32.0
+  - remote: buf.build/protocolbuffers/go:v1.32.0
```
Comment (pkwarren) [inline buf.gen.yaml:13]: Do you want to bump this to a newer version?

File: buf.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+lint:
+  use:
+    - DEFAULT
+  except:
+    - FIELD_NOT_REQUIRED
+    - PACKAGE_NO_IMPORT_CYCLE
+  ignore_only:
+    PROTOVALIDATE:
+      - proto/tests/example/v1/validations.proto
+  disallow_comment_ignores: true
```
Comment (pkwarren) [inline buf.yaml:17]: This is part of `buf config migrate` for compatibility sake, but I'd remove this setting and the `except:` setting to migrate to the defaults (as if you had created a new v2 config).

File: buf.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+breaking:
+  use:
+    - FILE
+  except:
```
Comment (pkwarren) [inline buf.yaml:21]: I'd also remove these to pick up the new v2 defaults.

File: buf.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  - buf.build/envoyproxy/protoc-gen-validate
+lint:
+  use:
+    - DEFAULT
+  except:
+    - FIELD_NOT_REQUIRED
+    - PACKAGE_NO_IMPORT_CYCLE
+  ignore_only:
+    PROTOVALIDATE:
+      - proto/tests/example/v1/validations.proto
```
Comment (pkwarren) [inline buf.yaml:16]: Is this meant to be relative to the module path or relative to the buf.yaml file?

File: buf.gen.yaml (hunk 1 above)
Comment (rodaine) [reply buf.gen.yaml:13]: Suppose I could since I'm in here

**Review by pkwarren — APPROVED** [review]

**Pull request merged**